test/cbmc/proofs/HTTPClient_AddHeader/Makefile

```diff
@@ -23,7 +23,17 @@ HARNESS_ENTRY=HTTPClient_AddHeader_harness
 PROOF_UID=HTTPClient_AddHeader
 HARNESS_FILE=$(HARNESS_ENTRY)
 
-DEFINES +=
+# The header field and value string lengths input into HTTPClient_AddHeader are
+# bounded to reduce the proof run time. This length is a bound on the strncpys
+# used to write the header field and values to the request header buffer. Memory
+# safety on the request header buffer can be proven in a reasonable bound. It
+# adds no value to the proof to write a header field and a header value of
+# SIZE_MAX length.
+HEADER_FIELD_MAX_LENGTH=16
+HEADER_VALUE_MAX_LENGTH=16
+
+DEFINES += -DHEADER_FIELD_MAX_LENGTH=$(HEADER_FIELD_MAX_LENGTH)
+DEFINES += -DHEADER_VALUE_MAX_LENGTH=$(HEADER_VALUE_MAX_LENGTH)
```

Comment:
This functions runs quickly without stubbing out strncpy.

Reply:
Do we have a rule of thumb for when we are allowed to bound the size of buffers? The user is always allowed to pass a header field or value that is larger than that new bound. It may be trivial to prove now, but a code change can easily invalidate these proofs.

Reply:
We bound everywhere that using SIZE_MAX would loop for SIZE_MAX increasing the run time of the proof. This is done in JSON, Shadow, Defender, and MQTT. Every library except this one.